File: adventofcode2018/1/p2.rs
use std::fs;
use std::collections::HashSet;

fn main() {
    let mut set: HashSet<i32> = vec!().into_iter().collect();
    let mut sum = 0;

    'outer: loop {
        for n in fs::read_to_string("input").expect("Unable to read file")
            .lines()
            .filter_map(|s| s.parse::<i32>().ok()) {
            sum += n;
            if set.contains(&sum) {
                break 'outer;
            }
            set.insert(sum);
        }
    }
    println!("{}", sum);
}

File: adventofcode/2018/2/p2.jl
file = open("input")
lines = readlines(file)

function solve(lines)
    for line1 in lines
        for line2 in lines
            if sum(collect(line1) .!= collect(line2)) == 1
                return collect(line1)[collect(line1) .== collect(line2)] |> join
            end
        end
    end
end

println(solve(lines))
